File: C++/Vertex3.cpp
#include "Vertex3.h"

Vertex3::Vertex3() : X(0.0f), Y(0.0f), Z(0.0f) { }

Vertex3::Vertex3(float x, float y, float z) {
    X = x;
    Y = y;
    Z = z;
}

Vertex3::~Vertex3() { }

float Vertex3::getX() { return X; }
float Vertex3::getY() { return Y; }
float Vertex3::getZ() { return Z; }
void Vertex3::setX(float x) { X = x; }
void Vertex3::setY(float y) { Y = y; }
void Vertex3::setZ(float z) { Z = z; }

void Vertex3::read(hsStream *S) {
    X = S->readFloat();
    Y = S->readFloat();
    Z = S->readFloat();
}

void Vertex3::write(hsStream *S) {
    S->writeFloat(X);
    S->writeFloat(Y);
    S->writeFloat(Z);
}

File: Java/Plasma/Vertex3.java
package Plasma;

import java.io.IOException;
import Plasma.CoreLib.hsStream;

public class Vertex3 {
    private float X, Y, Z;
    
    public Vertex3() {
        this(0.0f, 0.0f, 0.0f);
    }
    public Vertex3(float x, float y, float z) {
        X = x;
        Y = y;
        Z = z;
    }
    
    public float getX() { return X; }
    public float getY() { return Y; }
    public float getZ() { return Z; }
    public void setX(float x) { X = x; }
    public void setY(float y) { Y = y; }
    public void setZ(float z) { Z = z; }
    
    public void read(hsStream S) throws IOException {
        X = S.readFloat();
        Y = S.readFloat();
        Z = S.readFloat();
    }
    
    public void write(hsStream S) throws IOException {
        S.writeFloat(X);
        S.writeFloat(Y);
        S.writeFloat(Z);
    }
}
